src/solvers/strings/string_refinement.cpp

```diff
@@ -2060,6 +2060,9 @@ exprt string_refinementt::get(const exprt &expr) const
     }
     const auto array = supert::get(current.get());
     const auto index = get(index_expr->index());
+    if(array.is_nil())
```

Comment:
:pick: A comment explaining why we do that would be useful (the commit message is good in that sense).